script.js

```diff
@@ -1,4 +1,238 @@
+// Random Dice Roll Generator
+var randomDiceRoll = function () {
+  var randomDecimal = Math.random() * 6;
+  var randomInteger = Math.floor(randomDecimal);
+  var randomDiceNum = randomInteger + 1;
+  return randomDiceNum;
+};
+
+// Global Variables for Normal Dice Game
+var p1DiceOne = 0;
+var p1DiceTwo = 0;
+var p2DiceOne = 0;
+var p2DiceTwo = 0;
+var player1CombinedNum = 0;
+var player2CombinedNum = 0;
+p1DiceOne = randomDiceRoll();
+p1DiceTwo = randomDiceRoll();
+p2DiceOne = randomDiceRoll();
+p2DiceTwo = randomDiceRoll();
+
+// Global Variables for Variable Dice Game
+var gameMode = 'Enter Mode';
+var numOfDice = 0;
+var playerScore = 0;
+var player1Score = 0;
+var player2Score = 0;
+var totalp1Score = 0;
+var totalp2Score = 0;
+var p1DiceValues = [];
+var p1DicePosition = [];
+var p2DiceValues = [];
+var p2DicePosition = [];
+var dicePosition = [];
+var diceValues = [];
+
+// Helper function to list the indexes in the array
+var getDicePosition = function (diceLength) {
+  var index = 0;
+  while (index < diceLength) {
+    dicePosition.push(index);
+    index = index + 1;
+  }
+  return dicePosition;
+};
+
+// Helper function to list the random dice roll values
+var getNumOfDiceAndRoll = function (diceLength) {
+  var index = 0;
+  while (index < diceLength) {
+    diceValues.push(randomDiceRoll());
+    index = index + 1;
+  } return diceValues;
```

Comment:
Don't put code on the same line as a closing `}`, unless the code is the start of an `else if` or `else` block. Doing so makes the code harder to read, because `}` is typically on its own line by convention